packages/lib/services/plugins/PluginService.ts

```diff
@@ -410,19 +416,40 @@ export default class PluginService extends BaseService {
 
 			try {
 				const plugin = await this.loadPluginFromPath(pluginPath);
-
-				// After transforming the plugin path to an ID, multiple plugins might end up with the same ID. For
-				// example "MyPlugin" and "myplugin" would have the same ID. Technically it's possible to have two
-				// such folders but to keep things sane we disallow it.
-				if (this.plugins_[plugin.id]) throw new Error(`There is already a plugin with this ID: ${plugin.id}`);
+				const enabled = this.pluginEnabled(settings, plugin.id);
+
+				const existingPlugin = this.plugins_[plugin.id];
+				if (existingPlugin) {
+					const isSamePlugin = existingPlugin.baseDir === plugin.baseDir;
+
+					// On mobile, plugins can reload without restarting the app. If a plugin is currently
+					// running and hasn't changed, it doesn't need to be reloaded.
+					if (isSamePlugin) {
+						const isSameVersion =
+							existingPlugin.manifest.version === plugin.manifest.version
+							&& existingPlugin.manifest._package_hash === plugin.manifest._package_hash;
+						if (isSameVersion && existingPlugin.running === enabled) {
+							logger.debug('Not reloading same-version plugin', plugin.id);
+							continue;
+						} else {
+							logger.info('Reloading plugin with ID', plugin.id);
+							await this.unloadPlugin(plugin.id);
+						}
```

Comment:
This feels a bit messy, particularly because `.unloadPlugin` only works on mobile (and CLI). It could make sense to add something similar to the following to make it clear that this is the case:
```ts
if (!this.supportsUnloadingPlugins()) {
    throw new Error('some error message here');
}
```